### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by asyriac**

### CODE REVIEW

File: controllers/users.controller.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        return res.status(503).json({
+            success: false,
+            message : err.message
+        })
+    }
+}
+
+const login = async (req,res) => {
+    try{
+        const user = await User.findOne({"username":req.body.username})
```
Comment (Aravindan07) [inline controllers/users.controller.js]: Why username key is given as a string in 
```javascript
User.findOne(). User.findOne({username: req.body.username}) 
```
would work I think

File: db.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    mongoose.connect(`mongodb+srv://${process.env.DB_USERNAME}:${process.env.DB_PASSWORD}@cluster0.qtdp5.mongodb.net/flixcart?retryWrites=true&w=majority`,
+    {
+        useNewUrlParser: true,
+        useFindAndModify: false,
+        useUnifiedTopology: true,
+        useCreateIndex:true,
+    })
+    .then(()=> console.log('Connected to database'))
+    .catch((err)=>console.log("Connection to database failed",err))
+}
```
Comment (Aravindan07) [inline db.js]: Try to use try-catch instead of then and catch here

File: seed.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const faker = require('faker');
+const Product = require('./models/products.model')
+const initDB = require('./db')
+
+require('dotenv').config()
+initDB();
```
Comment (Aravindan07) [inline seed.js:6]: Just curious to know why you are initializing DB twice once in the index.js file and another one here.

**Review by Aravindan07 — CHANGES_REQUESTED** [review]
Hey, Arun you structured your backend code really well. I can't able to find any flaws or mistakes in your code, although I found some small changes it will not take more than half an hour to fix it. Good work mate!

**Review by sjsouvik — COMMENTED** [review]
Hey Arun,

The code is well structured and I couldn't find any issue with the code. Aravindan suggested some small changes, fix those and it'll be better. Good job!

File: seed.js (hunk 3 above)
Comment (asyriac) [reply seed.js:6]: This was just some stand alone code I wrote to populate the products collection. Not used as part of the app.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+.env
+/node_modules
```
Comment (khusharth) [inline .gitignore:2]: can add seed file here

**Pull request merged**